### ISSUE TITLE
Previous Element get read with Popover component on escape

### ISSUE THREAD

**Issue by Vinit-Pandit** [adobe/react-spectrum#9832]
### Provide a general summary of the issue here

When we close the popover dialog with `esc` it screenreader (NVDA ) read the previous interactive element while focus go to the triggered element.

### 🤔 Expected Behavior?

It should not read unnecessary element 

### 😯 Current Behavior

It is reading the previous interactive element on `esc` close of dialog

### 🔦 Context

https://github.com/user-attachments/assets/3ad0a6fc-68eb-434c-8c64-d3f95bf1eb70

### 🖥️ Steps to Reproduce

https://codesandbox.io/p/sandbox/nxd482?file=%2Fsrc%2FApp.js%3A13%2C23

https://github.com/user-attachments/assets/3ad0a6fc-68eb-434c-8c64-d3f95bf1eb70

### Version

Latest

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

Windows

**Comment by snowystinger** [comment]
I tried this with Windows 10.0.262000 and NVDA version 2025.3.3, I was unable to reproduce
